### PR TITLE
Register UI

### PR DESCRIPTION
**Pull request by karljerico**
**Overview**

- Created Register UI
- Initial route 
- Connect register to database

image
image

### CODE REVIEW

File: src/app/Http/Controllers/RegisterController.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+namespace App\Http\Controllers;
+
+use App\Models\User;
+use Illuminate\Http\Request;
+use Illuminate\Validation\Rule;
+
+class RegisterController extends Controller
+{
+    public function register(Request $request) {
+        $incomingFields = $request->validate([
```
Comment (hipe-francis) [inline src/app/Http/Controllers/RegisterController.php]: Please move this to a Form Request class.

File: src/routes/web.php [hunk 2]
```diff
@@ -14,5 +15,8 @@
 */
 
 Route::get('/', function () {
-    return view('welcome');
+    return view('register');
```
Comment (hipe-francis) [inline src/routes/web.php]: Better if you move this to the appropriate controller. Keep your route clean, and let your controllers handle your logics.

File: src/resources/views/register.blade.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      
+                        <div class="d-flex flex-row align-items-center mb-4">
+                          <i class="fas fa-lock fa-lg me-3 fa-fw"></i>
+                          <div class="form-outline flex-fill mb-0">
+                            <input name="password" type="password" id="form3Example4c" class="form-control" />
+                            <label class="form-label" for="form3Example4c">Password</label>
+                          </div>
+                        </div>
+      
+                        <!-- <div class="d-flex flex-row align-items-center mb-4">
```
Comment (hipe-francis) [inline src/resources/views/register.blade.php]: You can clean this up. Removed unused codes.

File: src/resources/views/register.blade.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </section>
+
+</body>
+</html>
```
Comment (bpocrafael) [inline src/resources/views/register.blade.php]: add new line

File: src/resources/views/register.blade.php (hunk 3 above)
Comment (hipe01mark) [reply src/resources/views/register.blade.php]: Just add. Use laravel blade template for Layout.

https://laravel.com/docs/10.x/blade

**Review by bpocrafael — COMMENTED** [review]
fix all the comments. Thanks

File: src/app/Http/Requests/RegisterRequest.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+use Illuminate\Validation\Rule;
+use Illuminate\Foundation\Http\FormRequest;
+
+
+class RegisterRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     */
+    public function authorize(): bool
```
Comment (bpocrafael) [inline src/app/Http/Requests/RegisterRequest.php]: do you think this method authorized is being used??

File: src/app/Http/Requests/RegisterRequest.php [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
+     */
+    public function rules(): array
+    {
+        return [
+            'name' => ['required', 'min:3', 'max:10', Rule::unique('users', 'name')],
+            'email' => ['required', 'email', Rule::unique('users', 'email')],
+            'password' => ['required', 'min:6', 'max:200'],
```
Comment (bpocrafael) [inline src/app/Http/Requests/RegisterRequest.php]: what about confirm password??

File: src/resources/views/layout/layout.blade.php [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<html lang="en">
+<head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.1/dist/css/bootstrap.min.css" rel="stylesheet">
+    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.1/dist/js/bootstrap.min.js"></script>
+
+</head>
```
Comment (bpocrafael) [inline src/resources/views/layout/layout.blade.php]: new line

File: src/resources/views/auth/register.blade.php [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </section>
+    
+</body>
```
Comment (bpocrafael) [inline src/resources/views/auth/register.blade.php]: hmmmm? 🤔

File: src/app/Http/Controllers/RegisterController.php [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace App\Http\Controllers;
+
+
+use App\Models\User;
+use App\Http\Requests\RegisterRequest;
+
+class RegisterController extends Controller
+{
+    public function view(){
```
Comment (bpocrafael) [inline src/app/Http/Controllers/RegisterController.php]: add return types

File: src/app/Http/Controllers/RegisterController.php [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+        return view('auth/register');
+    }
+
+    public function logout() {
```
Comment (bpocrafael) [inline src/app/Http/Controllers/RegisterController.php]: add return types

File: src/app/Http/Controllers/RegisterController.php [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+        auth()->logout();
+        return redirect('/');
+    }
+    
+    public function register(RegisterRequest $request)
```
Comment (bpocrafael) [inline src/app/Http/Controllers/RegisterController.php]: add return types

File: src/app/Http/Controllers/RegisterController.php (hunk 1 above)
Comment (karljerico) [reply src/app/Http/Controllers/RegisterController.php]: Created a commit for this comment, thank you Sir!

File: src/resources/views/register.blade.php (hunk 3 above)
Comment (karljerico) [reply src/resources/views/register.blade.php]: Created a commit for this comment, thank you Sir!

File: src/app/Http/Requests/RegisterRequest.php (hunk 5 above)
Comment (karljerico) [reply src/app/Http/Requests/RegisterRequest.php]: Already removed this method Sir, thank you!

File: src/app/Http/Requests/RegisterRequest.php (hunk 6 above)
Comment (karljerico) [reply src/app/Http/Requests/RegisterRequest.php]: Added a confirm password as well in the UI of the register Sir, Thank you!

File: src/resources/views/layout/layout.blade.php (hunk 7 above)
Comment (karljerico) [reply src/resources/views/layout/layout.blade.php]: Resolved this comment Sir, Thank you!

File: src/resources/views/auth/register.blade.php (hunk 8 above)
Comment (karljerico) [reply src/resources/views/auth/register.blade.php]: Resolved this comment Sir, Thank you!

File: src/app/Http/Controllers/RegisterController.php (hunk 9 above)
Comment (karljerico) [reply src/app/Http/Controllers/RegisterController.php]: Resolved this comment Sir, Thank you!

File: src/app/Http/Controllers/RegisterController.php (hunk 10 above)
Comment (karljerico) [reply src/app/Http/Controllers/RegisterController.php]: I removed the logout function for now Sir, Thank you!

File: src/app/Http/Controllers/RegisterController.php (hunk 11 above)
Comment (karljerico) [reply src/app/Http/Controllers/RegisterController.php]: Resolved this comment Sir, Thank you!

**Review by hipe-francis — APPROVED** [review]
LGTM 🚀

File: src/resources/views/auth/register.blade.php [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </section>
+
+@section('content')
```
Comment (hipe01mark) [inline src/resources/views/auth/register.blade.php:66]: Add new line

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

File: src/resources/views/auth/register.blade.php (hunk 12 above)
Comment (karljerico) [reply src/resources/views/auth/register.blade.php:66]: Added a new line Sir, Thank you!

**Review by hipe-francis — APPROVED** [review]
LGTM 🚀

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

**Review by bpocrafael — APPROVED** [review]
LGTM 🚀

**Review by hipe01mark — APPROVED** [review]
LGTM

**Review by nadj1233 — APPROVED** [review]
LGTM 💯

**Pull request merged**